core/src/components/datetime/datetime.tsx

```diff
@@ -2080,68 +2085,66 @@ export class Datetime implements ComponentInterface {
             }
 
             return (
-              <button
-                tabindex="-1"
-                data-day={day}
-                data-month={month}
-                data-year={year}
-                data-index={index}
-                data-day-of-week={dayOfWeek}
-                disabled={isCalDayDisabled}
-                class={{
-                  'calendar-day-padding': isCalendarPadding,
-                  'calendar-day': true,
-                  'calendar-day-active': isActive,
-                  'calendar-day-today': isToday,
-                }}
-                style={
-                  dateStyle && {
-                    color: dateStyle.textColor,
-                  }
-                }
-                aria-hidden={isCalendarPadding ? 'true' : null}
-                aria-selected={ariaSelected}
-                aria-label={ariaLabel}
-                onClick={() => {
-                  if (isCalendarPadding) {
-                    return;
+              <div class="calendar-day-wrapper">
+                <button
+                  tabindex="-1"
+                  data-day={day}
+                  data-month={month}
+                  data-year={year}
+                  data-index={index}
+                  data-day-of-week={dayOfWeek}
+                  disabled={isCalDayDisabled}
+                  class={{
+                    'calendar-day-padding': isCalendarPadding,
+                    'calendar-day': true,
+                    'calendar-day-active': isActive,
+                    'calendar-day-today': isToday,
+                  }}
+                  part={`calendar-day${isActive ? ' active' : ''}${isToday ? ' today' : ''}`}
```

Comment:
Should the parts be applied to disabled buttons?

![image](https://github.com/ionic-team/ionic-framework/assets/2721089/6926c34f-12d0-4a26-94a0-6bfa2bffcab8)

Reply:
I think it should still apply to disabled, since disabled is a state. If anything, we could add another part for `disabled` so that they could target that part specifically.

Reply:
Ah sorry I misspoke. I was referring more to the spacer elements (see the two disabled buttons to the left of the "1" button). I do agree that this shadow part should apply to a disabled day button (i.e. if the "1" button was disabled).

Reason I ask is the spacer elements are there for alignment purposes and aren't supposed to represent calendar days

Reply:
The part should not apply to the spacer elements. For the highlighting dates API we deliberately do not run it for the spacer elements: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/datetime/datetime.tsx#L2078-L2080 (`day !== null`, could of used the `isCalendarPadding` const as well).

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/27641/commits/d522a7661a439e376509f096c7d9399b973f4b1e. Will run screenshot tests soon that add a background to show that they aren't styled, just waiting to make a decision on: https://github.com/ionic-team/ionic-framework/pull/27641#discussion_r1303169699